keps/sig-network/4858-ip-cidr-validation/README.md

```diff
@@ -0,0 +1,838 @@
+enforcing tighter validation for new objects of existing types means
+that we might break some users' existing workflows / automation that
+were generating now-invalid values. This will be mitigated by having
+API warnings for a few releases before we flip the switch, and by
+wrapping the new validation in a feature gate that users could disable
+if needed while they update their infrastructure.
+
+The new validation logic (to allow legacy invalid values) will be more
+complicated than the old logic, and thus potentially more likely to
+have bugs.
+
+## Design Details
+
+Assuming all of the UNRESOLVED sections are ignored, most of the
+updated validation (not including a feature gate) is already
+implemented, in [kubernetes #122550] and [kubernetes #128786]; the KEP
+is being written retroactively to make sure we have agreement on that
+plan (or not).
+
+[kubernetes #122550]: https://github.com/kubernetes/kubernetes/pull/122550
+[kubernetes #128786]: https://github.com/kubernetes/kubernetes/pull/128786
+
+### Test Plan
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+The already-merged changes to make IP/CIDR validation more consistent
+also included updates to unit tests both for the generic functions
+(e.g. `utilvalidation.IsValidIP`) and for some specific fields (e.g.
+the weird edge cases of `LoadBalancerSourceRanges`.)
+
+##### Unit tests
+
+We will add tests for the new IP/CIDR warning validation functions,
+and for the fields that use them. We will test the behavior of being
+able to make unrelated changes to objects with pre-existing invalid
+values, and the ability to fix invalid fields of immutable objects, to
+the extent agreed on above.
+
+- `k8s.io/apimachinery/pkg/util/validation`: `2024-10-29` - `82.3`
+- `k8s.io/kubernetes/pkg/api/pod/warnings.go`: `2024-10-29` - `97.4`
+- `k8s.io/kubernetes/pkg/api/service/warnings.go`: `2024-10-29` - `96.8`
+- `k8s.io/kubernetes/pkg/apis/core/validation`: `2024-10-29` - `84.4`
+- `k8s.io/kubernetes/pkg/apis/discovery/validation`: `2024-10-29` - `98.8`
+- `k8s.io/kubernetes/pkg/apis/networking/validation`: `2024-10-29` - `91.4`
+- `k8s.io/kubernetes/pkg/registry/core/endpoint`: `2024-10-29` - `0.0`
+- `k8s.io/kubernetes/pkg/registry/core/node`: `2024-10-29` - `50.0`
+- `k8s.io/kubernetes/pkg/registry/core/pod`: `2024-10-29` - `64.0`
+- `k8s.io/kubernetes/pkg/registry/core/service`: `2024-10-29` - `73.8`
+- `k8s.io/kubernetes/pkg/registry/discovery/endpointslice`: `2024-10-29` - `79.2`
+- `k8s.io/kubernetes/pkg/registry/networking/ingress`: `2024-10-29` - `64.5`
+- `k8s.io/kubernetes/pkg/registry/networking/networkpolicy`: `2024-10-29` - `76.9`
+
+##### Integration tests
+
+No new tests, and we will remove
+`test/integration/apiserver/cve_2021_29923_test.go`, which tests that
```

Comment:
I'd expect this test to exist until we lock the feature gate, and its execution should be guarded with the feature being off.